File: java_integration/app/src/main/java/com/firewok/example/java_integration/feedresource/singlecontent/SingleContentActivity.java
package com.firewok.example.java_integration.feedresource.singlecontent;

import android.annotation.SuppressLint;
import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.MenuItem;
import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import com.firework.common.feed.FeedResource;
import com.firewok.example.java_integration.BuildConfig;
import com.firewok.example.java_integration.R;
import com.firewok.example.java_integration.databinding.ActivitySingleContentBinding;
import com.firework.videofeed.FwVideoFeedView;
import com.firework.viewoptions.BaseOption;
import com.firework.viewoptions.ViewOptions;

public class SingleContentActivity extends AppCompatActivity {
    private ActivitySingleContentBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivitySingleContentBinding.inflate(LayoutInflater.from(this));
        setContentView(binding.getRoot());
        
        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.setTitle(getString(R.string.single_content_screen_title));
            actionBar.setDisplayHomeAsUpEnabled(true);
            actionBar.setDisplayShowHomeEnabled(true);
        }

        setupDetails();
        initVideoFeedView();
    }

    @SuppressLint("SetTextI18n")
    private void setupDetails() {
        binding.details.source.setText("SingleContent");
        binding.details.elementId.setText(BuildConfig.FW_CONTENT_ID);
    }

    private void initVideoFeedView() {
        FwVideoFeedView videoFeedView = binding.fwVideoFeedView;

        ViewOptions viewOptions = new ViewOptions.Builder()
            .baseOption(new BaseOption.Builder()
                .feedResource(new FeedResource.SingleContent(BuildConfig.FW_CONTENT_ID))
                .build())
            .build();

        videoFeedView.init(viewOptions);
    }

    @Override
    protected void onDestroy() {
        binding.fwVideoFeedView.destroy();
        super.onDestroy();
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == android.R.id.home) {
            finish();
            return true;
        }
        return super.onOptionsItemSelected(item);
    }

    public static Intent getIntent(Context context) {
        return new Intent(context, SingleContentActivity.class);
    }
} 

File: feed_resources/app/src/main/java/com/firework/example/feedresources/singlecontent/SingleContentActivity.kt
package com.firework.example.feedresources.singlecontent

import android.annotation.SuppressLint
import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import androidx.appcompat.app.AppCompatActivity
import com.firework.common.feed.FeedResource
import com.firework.example.feedresources.BuildConfig.FW_CONTENT_ID
import com.firework.example.feedresources.R
import com.firework.example.feedresources.databinding.ActivitySingleContentBinding
import com.firework.viewoptions.baseOptions
import com.firework.viewoptions.viewOptions

class SingleContentActivity : AppCompatActivity() {

    private lateinit var binding: ActivitySingleContentBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivitySingleContentBinding.inflate(LayoutInflater.from(this))
        setContentView(binding.root)
        setTitle(R.string.single_content_screen_title)

        setupDetails()

        initVideoFeedView()
    }

    @SuppressLint("SetTextI18n")
    private fun setupDetails() {
        binding.details.source.text = "SingleContent"
        binding.details.elementId.text = FW_CONTENT_ID
    }

    private fun initVideoFeedView() {
        val videoFeedView = binding.fwVideoFeedView

        val viewOptions = viewOptions {
            baseOptions {
                // Check Channel, Playlist, or Dynamic screens for other feed sources
                feedResource(FeedResource.SingleContent(FW_CONTENT_ID))
            }
        }

        videoFeedView.init(viewOptions)
    }

    override fun onDestroy() {
        binding.fwVideoFeedView.destroy()
        super.onDestroy()
    }

    companion object {
        fun intent(context: Context) = Intent(context, SingleContentActivity::class.java)
    }
}
